app/src/main/java/org/falaeapp/falae/database/DownloadCacheDbHelper.kt

```diff
@@ -0,0 +1,106 @@
+package org.falaeapp.falae.database
+
+import android.content.ContentValues
+import android.content.Context
+import android.util.Log
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import com.google.gson.reflect.TypeToken
+import org.falaeapp.falae.model.DownloadCache
+
+
+/**
+ * Created by marce on 03/02/2018.
+ */
+class DownloadCacheDbHelper(context: Context) {
+
+    private val databaseHelper: DataBaseHelper = DataBaseHelper.getInstance(context)
+    private var gson = GsonBuilder().create()
+
+    class DownloadCacheEntry {
+        companion object {
+
+            const val TABLE_NAME = "download_cache"
+            const val _ID = "_id"
+            const val COLUMN_NAME = "name"
+            const val COLUMN_SOURCES = "sources"
+            const val _COUNT = "_count"
+        }
+    }
+
+    fun insert(downloadCache: DownloadCache): Long {
+        val json = gson.toJson(downloadCache.sources)
+        val contentValues = ContentValues()
+        contentValues.put(DownloadCacheEntry.COLUMN_NAME, downloadCache.name)
+        contentValues.put(DownloadCacheEntry.COLUMN_SOURCES, json)
+        val db = databaseHelper.writableDatabase
+        Log.d(javaClass.name, "Inserting data...")
+        return db.insert(DownloadCacheEntry.TABLE_NAME, null, contentValues)
+    }
+
+
+    fun update(downloadCache: DownloadCache) {
+        val json = gson.toJson(downloadCache.sources)
+        val contentValues = ContentValues()
+        contentValues.put(DownloadCacheEntry.COLUMN_NAME, downloadCache.name)
+        contentValues.put(DownloadCacheEntry.COLUMN_SOURCES, json)
+        val db = databaseHelper.writableDatabase
+        Log.d(javaClass.name, "Updating entry...")
+        db.update(DownloadCacheEntry.TABLE_NAME, contentValues,
+                DownloadCacheDbHelper.DownloadCacheEntry.COLUMN_NAME + "= ? ",
```

Comment:
For better readability, Make/use selection query as local var. Do the same for selection args below.